// Generated by CoffeeScript 1.10.0
(function() {
  var Deux, Remote, Test, expose, rem2, remote, src,
    slice = [].slice;

  Remote = require('avs-easyrpc').Remote;

  expose = require('avs-easyrpc').expose;

  module.exports = remote = new Remote({
    "class": 'Employee',
    methods: ['getProfile', 'speak']
  });

  Test = (function() {
    function Test() {}

    Test.prototype.echo = function() {
      var text, user;
      user = arguments[0], text = 2 <= arguments.length ? slice.call(arguments, 1) : [];
      console.log.apply(console, [user + ":"].concat(slice.call(text)));
      $('#messages').append($('<li>').text(user + ":" + text[0]));
      return null;
    };

    return Test;

  })();

  expose(new Test(), remote).then(function() {
    return remote.speak('hello');
  });

  remote.prespeak = function() {
    remote.speak($('#m').val());
    return $('#m').val('');
  };

  src = null;

  Deux = (function() {
    function Deux() {}

    Deux.prototype.test = function(msg) {
      console.log('deux: ', msg);
      src.close();
      return null;
    };

    return Deux;

  })();

  expose(new Deux(), new Remote({
    "class": 'Customer'
  })).then(function(source) {
    return src = source;
  });

  rem2 = new Remote({
    "class": 'doesnotexist',
    methods: ['test']
  });

  rem2.test().then(function(rep) {
    return console.log(rep);
  })["catch"](function(err) {
    return console.log(err);
  });


  /*
  remote.getProfile 'john'
  .then (rep) -> console.log rep
  .catch (err) -> console.log err
  
  remote.publish()
  .then (rep) -> console.log rep
  .catch (err) -> console.log err
   */

}).call(this);
